app/controllers/users_controller.rb

```diff
@@ -8,6 +8,22 @@ def show
       @location = Location.find(@user.location_id)
     end
     @favorite_discussions = @user.favorite_discussions
+    if @user.is_admin
+      @pending_map_users = User.where(map_approval_state: 0, on_map: true)
+    else
+      @pending_map_users = []
+    end
+  end
+
+  def approval_update
+    # batch update user's map_approval_state based on admin decision
+    if @user.is_admin
```

Comment:
you need to define @user somewhere before you use it. Try passing in the admin's admin_id as another param when you make the request and then do User.find(params[:admin_id])